cli/src/test/java/com/devonfw/tools/ide/io/FileAccessImplTest.java

```diff
@@ -0,0 +1,299 @@
+
+    fa.symlink(dir.resolve("d1/d11"), dir.resolve("d1/d11/link2"), relative);
+    fa.symlink(dir.resolve("d1/d11/d111"), dir.resolve("d1/d11/link3"), relative);
+    fa.symlink(dir.resolve("d1/d11/d111/d1111"), dir.resolve("d1/d11/link4"), relative);
+    fa.symlink(dir.resolve("d2"), dir.resolve("d1/d11/link5"), relative);
+    fa.symlink(dir.resolve("d2/d22"), dir.resolve("d1/d11/link6"), relative);
+    fa.symlink(dir.resolve("d2/d22/d222"), dir.resolve("d1/d11/link7"), relative);
+
+    fa.symlink(dir.resolve("d1/d11/link1"), dir.resolve("d2/d22/link8"), relative);
+    fa.symlink(dir.resolve("d1/d11/link1"), dir.resolve("d2/link9"), relative);
+    fa.symlink(dir.resolve("d2/link9"), dir.resolve("link10"), relative);
+  }
+
+  private void assertSymlinksExist(Path dir) {
+
+    assertThat(dir.resolve("d1/d11/link1")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link2")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link3")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link4")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link5")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link6")).existsNoFollowLinks();
+    assertThat(dir.resolve("d1/d11/link7")).existsNoFollowLinks();
+    assertThat(dir.resolve("d2/d22/link8")).existsNoFollowLinks();
+    assertThat(dir.resolve("d2/link9")).existsNoFollowLinks();
+    assertThat(dir.resolve("link10")).existsNoFollowLinks();
+  }
+
+  private void assertSymlinksAreBroken(Path dir, boolean readLinks) throws IOException {
+
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link1"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link2"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link3"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link4"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link5"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link6"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d1/d11/link7"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d2/d22/link8"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("d2/link9"), readLinks);
+    assertSymlinkIsBroken(dir.resolve("link10"), readLinks);
+  }
+
+  private void assertSymlinkIsBroken(Path link, boolean readLinks) throws IOException {
+
+    try {
+      Path realPath = link.toRealPath();
+      if (Files.exists(realPath)) {
+        fail("The link target " + realPath + " (from toRealPath) should not exist");
+      }
+    } catch (IOException e) { // toRealPath() throws exception for junctions
+      assertThat(e).isInstanceOf(NoSuchFileException.class);
+    }
+    if (readLinks) {
+      Path readPath = Files.readSymbolicLink(link);
+      if (Files.exists(readPath)) {
+        fail("The link target " + readPath + " (from readSymbolicLink) should not exist");
+      }
+    }
+  }
+
+  /*
+   * only pass readLinks = true when junctions are not used.
+   */
```

Comment:
If you want to use JavaDoc start with `/**` and use `@param readLinks` to be more precise.
Otherwise if you only want a code-comment simply use `//` what avoids two pointless extra lines ;)